### DISCUSSION TITLE
Change select list of embedded resource without changing endpoint URL

### DISCUSSION THREAD

**Discussion by BorislavZlatanov** [PostgREST/postgrest#3158] [Q&A] [answered]
Currently, I have resource embedding using an endpoint call such as this one:

`/rpc/polls?select=*,poll_choices(choice_num,choice_text,poll_votes(hp:hp.sum(),spsp:spsp.sum()))`

This returns the results from my polls table, with embedded results from my poll_choices table and within that embedded results from my poll_votes table. It is working. But I expect to be adding more fields to the poll_votes() select list over time, so I tried to make a view poll_votes_view that selects only the needed fields (and I'd be adding more fields to it over time). But the resource embedding is not working - it gives "Could not find a relationship between 'poll_choices' and 'poll_votes_view' in the schema cache". It works only if I do `SELECT *` in the view, which makes the view pointless.

How can I use a view (or any other approach) in this situation so that I can expand the select list of poll_votes without needing to change the endpoint URL?

**Comment by laurenceisla** [comment, marked as answer]
That's because resource embedding  in views require that:

> The foreign keys’ columns must be present in the top SELECT clause of the view for this to work.

No need to select all the columns, but the `poll_choices` id **must** be present. See [Foreign Key Joins on Views in the docs](https://postgrest.org/en/v12.0/references/api/resource_embedding.html#foreign-key-joins-on-views).

**Reply by BorislavZlatanov** [reply]
Thanks, guess I didn't understand it too well in the documentation. 

Managed to get it working with a view `votes` that has in its SELECT list the columns that the `poll_choices` and the `poll_votes` foreign_keys point to. So this made the connections between my `polls`, `poll_choices` and `poll_votes` tables, and I was able to make the endpoint like this:

`/rpc/polls?select=*,votes(*))`

So I can now add more columns to the `votes` view without needing to change the endpoint. Thanks again.
